components/runners/ambient-runner/ambient_runner/observability.py

```
@@ -521,6 +535,14 @@ def end_turn(
             if self._current_turn_ctx:
                 self._current_turn_ctx.__exit__(None, None, None)
 
+            # Exit per-turn propagation context
+            if self._turn_propagate_ctx:
+                try:
+                    self._turn_propagate_ctx.__exit__(None, None, None)
+                except Exception:
+                    pass
```

Ruff (0.15.6)

[warning] Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(Exception): ...`

(SIM105)